test/kubemark/pre-existing/util.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function run-cmd-with-retries {
+  RETRIES="${RETRIES:-3}"
+  for attempt in $(seq 1 ${RETRIES}); do
+    exec 5>&1
+    result=$("$@" |& tee >(cat - >&5)) || local ret_val="$?"
+    echo "${result}"
+    if [[ "${ret_val:-0}" -ne "0" ]]; then
+      if [[ $(echo "${result}" | grep -c "already exists") -gt 0 ]]; then
+        if [[ "${attempt}" == 1 ]]; then
+          echo -e "${color_red}Failed to $1 $2 $3 as the resource hasn't been deleted from a previous run.${color_norm}" >& 2
+          exit 1
+        fi
+        echo -e "${color_yellow}Succeeded to $1 $2 $3 in the previous attempt, but status response wasn't received.${color_norm}"
+        return 0
+      fi
+      echo -e "${color_yellow}Attempt $attempt failed to $1 $2 $3. Retrying.${color_norm}" >& 2
+      sleep $(($attempt * 5))
+    else
+      echo -e "${color_green}Succeeded to gcloud compute $1 $2 $3.${color_norm}"
+      return 0
+    fi
+  done
+  echo -e "${color_red}Failed to $1 $2 $3.${color_norm}" >& 2
+  exit 1
+}
+
+# Leave the skeleton definition of execute-cmd-on-master-with-retries
+# so only the pre-existing provider functions will target this.
+function execute-cmd-on-pre-existing-master-with-retries() {
```

Comment:
The function ```execute-cmd-on-master-with-retries``` will be passed because I don't want to write anything to the master.  I only need to read the existing cluster credentials. I commented at the bottom to further explain.